core/domain/classifier_domain.py

```diff
@@ -504,7 +474,126 @@ def validate(self):
                 'Expected state_name to be a string, received %s' % (
                     self.state_name))
 
-        if not isinstance(self.job_id, python_utils.BASESTRING):
+        self._algorithm_ids_to_job_ids.validate()
+
+
+class AlgorithmIdToJobIdMapping(python_utils.OBJECT):
```

Comment:
Sorry, I think there might have been a misunderstanding. I don't think this domain object is going to work so well -- I keep confusing it with the dict when reading the code. It would help if they were named more differently, but I can't think of a better name. But it's also really just a wrapper for a dict, and not really an object, conceptually.

So I suggest just using the dict instead. Just make sure to have a very clear explanation of what's in the dict when introducing it in the args/etc. docstring.

Reply:
Done